addons/mail/models/discuss/discuss_channel.py

```diff
@@ -1428,16 +1428,21 @@ def channel_fetched(self):
             """, member_query))
             members = cursor_self.env['discuss.channel.member'].browse([r[0] for r in to_update])
             channel2message = members.channel_id._get_last_messages().grouped('channel_id')
+            updated_members_by_channel = defaultdict(cursor_self.env["discuss.channel.member"].browse)
             for member in members:
                 last_message = channel2message[member.channel_id]
                 if member.fetched_message_id != last_message:
                     member.fetched_message_id = last_message
-                    member.channel_id._bus_send('discuss.channel.member/fetched', {
-                        'channel_id': member.channel_id.id,
-                        'id': member.id,
-                        'last_message_id': last_message.id,
-                        'partner_id': cursor_self.env.user.partner_id.id,
-                    })
+                    updated_members_by_channel[member.channel_id] |= member
+            for channel in updated_members_by_channel:
+                Store(bus_channel=channel).add(
+                    updated_members_by_channel[channel],
+                    fields=[
+                        Store.One("channel_id", []),
+                        "fetched_message_id",
+                        Store.One("partner_id", []),
```

Comment:
Ideally this should be `_to_store_persona` (with no fields) instead to properly take guests into account (they currently don't have a fetch feature but there should be no limitation to allow it).